WordPress/src/main/java/org/wordpress/android/ui/posts/PromoDialogAdvanced.java

```diff
@@ -0,0 +1,122 @@
+package org.wordpress.android.ui.posts;
+
+import android.app.Dialog;
+import android.os.Bundle;
+import android.support.annotation.DrawableRes;
+import android.support.annotation.NonNull;
+import android.support.annotation.StringRes;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Button;
+import android.widget.ImageView;
+import android.widget.LinearLayout;
+
+import org.wordpress.android.R;
+import org.wordpress.android.util.DisplayUtils;
+import org.wordpress.android.widgets.WPTextView;
+
+/**
+ * Similar to {@link PromoDialog}, but with an optional link field and negative button.
+ */
+public class PromoDialogAdvanced extends PromoDialog {
+    public static class Builder extends PromoDialog.Builder {
+        @StringRes int linkId;
+        @StringRes int buttonNegativeId;
+
+        public Builder(@DrawableRes int drawableId, @StringRes int titleId, @StringRes int descriptionId,
+                       @StringRes int buttonPositiveId) {
+            super(drawableId, titleId, descriptionId, buttonPositiveId);
+        }
+
+        public Builder setLinkText(@StringRes int linkId) {
+            this.linkId = linkId;
+            return this;
+        }
+
+        public Builder setNegativeButtonText(@StringRes int buttonNegativeId) {
+            this.buttonNegativeId = buttonNegativeId;
+            return this;
+        }
+
+        @Override
+        public PromoDialogAdvanced build() {
+            return PromoDialogAdvanced.newInstance(this);
+        }
+    }
+
+    protected int mButtonNegativeId;
+    protected int mLinkId;
+
+    protected View.OnClickListener mNegativeButtonOnClickListener;
+    protected View.OnClickListener mLinkOnClickListener;
+
+    protected static PromoDialogAdvanced newInstance(Builder builder) {
+        PromoDialogAdvanced fragment = new PromoDialogAdvanced();
+        Bundle args = new Bundle();
+        args.putInt("drawableId", builder.drawableId);
```

Comment:
Can we move the "drawableId", "titleId", etc strings to `private static final String...`?